src/python_testing/TC_CADMIN_1_22.py

```diff
@@ -0,0 +1,140 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ALL_CLUSTERS_APP}
+# test-runner-run/run1/factoryreset: True
+# test-runner-run/run1/quiet: True
+# test-runner-run/run1/app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+# test-runner-run/run1/script-args: --storage-path admin_storage.json --commissioning-method on-network --discriminator 1234 --passcode 20202021 --trace-to json:${TRACE_TEST_JSON}.json --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto --PICS src/app/tests/suites/certification/ci-pics-values
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import random
+from time import sleep
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.ChipDeviceCtrl import CommissioningParameters
+from chip.exceptions import ChipStackError
+from chip.native import PyChipError
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_CADMIN_1_22(MatterBaseTest):
+    async def OpenCommissioningWindow(self) -> CommissioningParameters:
+        try:
+            params = await self.th1.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=self.max_window_duration, iteration=10000, discriminator=self.discriminator, option=1)
+            return params
+
+        except Exception as e:
+            logging.exception('Error running OpenCommissioningWindow %s', e)
+            asserts.assert_true(False, 'Failed to open commissioning window')
+
+    def generate_unique_random_value(self, value):
+        while True:
+            random_value = random.randint(10000000, 99999999)
+            if random_value != value:
+                return random_value
+
+    async def CommissionAttempt(
+            self, setupPinCode: int, thnum: int, th: str, fail: bool):
+
+        logging.info(f"-----------------Commissioning with TH_CR{str(thnum)}-------------------------")
+        if fail:
+            ctx = asserts.assert_raises(ChipStackError)
+            self.print_step(0, ctx)
+            with ctx:
+                await th.CommissionOnNetwork(
+                    nodeId=self.dut_node_id, setupPinCode=setupPinCode,
+                    filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.discriminator)
+                errcode = PyChipError.from_code(ctx.exception.err)
+            logging.info('Commissioning complete done. Successful? {}, errorcode = {}'.format(
+                errcode.is_success, errcode))
+            asserts.assert_false(errcode.is_success, 'Commissioning complete did not error as expected')
+            asserts.assert_true(errcode.sdk_code == 0x0000000B,
```

Comment:
is there a codegen enum value you could use here? It seems like these values should appear in the enums section of this cluster

Reply:
Sorry this function that is mentioned above is not in these tests, this code for that function has been removed in this test.
However, I agree that the enum value for the errcode.sdk_code should have an associated value with its cluster, I will find it and replace it in the other tests such as CADMIN_1_3.